clang/lib/Sema/SemaChecking.cpp

```diff
@@ -168,7 +168,7 @@ bool Sema::checkArgCount(CallExpr *Call, unsigned DesiredArgCount) {
 
   return Diag(Range.getBegin(), diag::err_typecheck_call_too_many_args)
          << 0 /*function call*/ << DesiredArgCount << ArgCount
-         << /*is non object*/ 0 << Call->getArg(1)->getSourceRange();
+         << /*is non object*/ 0 << Range;
```

Comment:
The wasm stuff LGTM, but I don't really know anything about SemaChecking.cpp so someone else might have to review this?

Reply:
It seems to have fixed a real bug in the labeling:
```C
$ clang -c a.c 
a.c:2:30: error: too many arguments to function call, expected 2, have 4
    2 |   __builtin_annotation(1, 2, 3, 4);
      |                           ~  ^
1 error generated.
$ ./bin/clang -c a.c 
a.c:2:30: error: too many arguments to function call, expected 2, have 4
    2 |   __builtin_annotation(1, 2, 3, 4);
      |                              ^~~~
1 error generated.
```

Reply:
Should I split this change into a separate PR?

Reply:
That sounds like a good idea yes.